doc/source/dev/plugins.rst

```diff
@@ -0,0 +1,88 @@
+.. devguide_plugins:
+
+Developing Plugins
+------------------
+
+As discussed in the user's guide, WATTS has a variety of plugins for different
+codes (primarily focused on application codes in nuclear science and
+engineering) that are already available. If you wish to use WATTS with a code
+that does not already have a plugin available, extending WATTS to handle another
+code can be done quite easily.
+
+There are two ways to incorporate new codes into WATTS. First, if the code uses
+a text-based input, the :class:`~watts.PluginTemplate` class can be used to
+specify the executable, how it should be invoked from a command-line, and a
+template for the input file. For example, if you have a code called ``goblin``
+that should be executed from a command line as:
+
+.. code-block:: sh
+
+    goblin input=filein.gbl
+
+a plugin can be set up as::
+
+    goblin = watts.PluginTemplate(
+        executable='goblin',
+        execute_command=['{self.executable}', 'input={self.input_name}'],
+        template_file='filein.gbl'
+    )
+
+where ``filein.gbl`` is a template input for the ``goblin`` code. Executing the
+code is done by passing it a set of parameters::
+
+    params = watts.Parameters()
+    ...
+
+    result = goblin(params)
```

Comment:
it is not clear to me what results will be passed from the template. We may want to specify here what to expect in results when using the standard plugin.